### PR TITLE
Touch-Friendly UI Tweaks (Separate small-screen UI?)

### PR DESCRIPTION
**Pull request by SomeInterestingUserName**
I've found gqrx running on a Raspberry Pi touchscreen to be a nice and portable RF spectrum analyzer. However, those displays generally don't have much screen area, and don't seem to play particularly nicely with the current desktop-centric interface. I've been working on some UI tweaks to make the interface more touch-friendly, but I'm not sure if they would be applicable to larger screens/non-touch screens. Perhaps a separate branch aimed at low-res/portable displays is in order?

So far, the tweaks I've made allow the settings docks to be hidden (to give the FFT/waterfall more screen space) and make it easier to see where to tap on the frequency control.

### CODE REVIEW

**Comment by csete** [comment]
I will take a look at it when I have some time. In the meanwhile, you are welcome to post some screenshots that illustrate what this does.

**Comment by SomeInterestingUserName** [comment]
So far, the changes are:

menu_minimized
^ A button to minimize the settings dock on the right-hand side of the screen. While the space savings isn't too apparent on a large screen, the dock takes up nearly 50% of a 7" Raspberry Pi touch screen, making the FFT and waterfall quite cramped.

menu_normal
^ The menus can be restored via a new button on the toolbar (The "i" button). This will probably need a more descriptive logo in the future. 

freqcontrol
^ The frequency control now faintly shows leading zeroes, making it easier to see where to click in order to adjust a certain digit. 

That's what I have so far in this commit. I had a couple more features in mind, but they're more device-specific.

File: src/applications/gqrx/mainwindow.cpp [hunk 1]
```diff
@@ -464,6 +467,10 @@ bool MainWindow::loadConfig(const QString cfgfile, bool check_crash,
         restoreState(m_settings->value("gui/state", saveState()).toByteArray());
     }
 
+    // Restore settings menu hide/show state
+    bool_val = m_settings->value("gui/showmenu", false).toBool();
```
Comment (csete) [inline src/applications/gqrx/mainwindow.cpp]: "showmenu" is a misleading name since these widgets are control panels and now menus. Furthermore, the default value should be true (visible) since that's how the program works now.

File: src/applications/gqrx/mainwindow.cpp [hunk 2]
```diff
@@ -2300,3 +2307,20 @@ void MainWindow::on_actionAddBookmark_triggered()
...
+{
+    if(state){
+        if(stateBeforeMinimize != Q_NULLPTR)
+            restoreState(stateBeforeMinimize);
+    }
+    else{
+        //Save what the dock area looks like, then hide all dock widgets
+        stateBeforeMinimize = saveState();
+        for(QDockWidget* dw: findChildren<QDockWidget *>()){
+            if(dockWidgetArea(dw) == Qt::RightDockWidgetArea)
```
Comment (csete) [inline src/applications/gqrx/mainwindow.cpp]: The placement of the dock widgets is not limited to the right side of the window; however, the implementation will only work for dock widgets placed on the right side of the main window.

File: src/applications/gqrx/mainwindow.cpp [hunk 3]
```diff
@@ -2300,3 +2307,20 @@ void MainWindow::on_actionAddBookmark_triggered()
         ui->plotter->updateOverlay();
     }
 }
+
+// Hide/show the right-hand settings menu, to give a larger FFT viewing area
+void MainWindow::on_actionShowRightDock_toggled(bool state)
+{
+    if(state){
```
Comment (csete) [inline src/applications/gqrx/mainwindow.cpp]: Please observe and follow the coding style of the surrounding code.

File: src/applications/gqrx/mainwindow.h [hunk 4]
```diff
@@ -121,6 +121,8 @@ public slots:
     // dummy widget to enforce linking to QtSvg
     QSvgWidget      *qsvg_dummy;
 
+    QByteArray stateBeforeMinimize;
```
Comment (csete) [inline src/applications/gqrx/mainwindow.h]: The name should reflect whos state we are talking about here. I'm not sure why "minimize"?

File: src/applications/gqrx/mainwindow.ui [hunk 5]
```diff
@@ -541,6 +542,25 @@
...
+   <property name="checkable">
+    <bool>true</bool>
+   </property>
+   <property name="checked">
+    <bool>true</bool>
+   </property>
+   <property name="icon">
+    <iconset resource="../../../resources/icons.qrc">
+     <normaloff>:/icons/icons/info.svg</normaloff>
+     <normalon>:/icons/icons/info.svg</normalon>:/icons/icons/info.svg</iconset>
```
Comment (csete) [inline src/applications/gqrx/mainwindow.ui]: I don't think the Info icon is suitable for this purpose.

File: src/applications/gqrx/mainwindow.ui [hunk 6]
```diff
@@ -541,6 +542,25 @@
...
+   </property>
+   <property name="text">
+    <string>Show Menus</string>
```
Comment (csete) [inline src/applications/gqrx/mainwindow.ui]: These are control panel, not menus.

File: src/applications/gqrx/mainwindow.ui [hunk 7]
```diff
@@ -86,8 +86,8 @@
          </property>
          <property name="sizeHint" stdset="0">
           <size>
-           <width>5</width>
-           <height>15</height>
+           <width>27</width>
```
Comment (csete) [inline src/applications/gqrx/mainwindow.ui]: Not sure what these changes are; seem unrelated to the topic."

**Comment by csete** [comment]
I have merged the changes to the frequency control widget manually; however, the show/hide control panels are too problematic in my opinion, see my comments. Feel free to submit a new pull request if you resolve those issues.
